File: frontend/types/doctor.ts
export interface Doctor {
  id: string
  specialization: string | null
  department: string | null
  bio: string | null
  consultation_fee: number | null
  languages_spoken: string[]
  slot_duration_minutes: number
  buffer_minutes: number
  is_active: boolean
  user: { id: string; name: string; email: string; phone: string | null; avatar_url: string | null; role: string; is_active: boolean; created_at: string }
}

export interface DoctorPublicProfile {
  id: string
  specialization: string | null
  department: string | null
  bio: string | null
  consultation_fee: number | null
  languages_spoken: string[]
  slot_duration_minutes: number
  name: string
  avatar_url: string | null
}

export interface DoctorUpdate {
  specialization?: string
  department?: string
  bio?: string
  consultation_fee?: number
  languages_spoken?: string[]
  slot_duration_minutes?: number
  buffer_minutes?: number
}

File: backend/app/schemas/doctor.py
from uuid import UUID
from decimal import Decimal
from pydantic import BaseModel
from app.schemas.user import UserResponse


class DoctorCreate(BaseModel):
    # User fields
    name: str
    email: str
    phone: str | None = None
    password: str
    # Doctor fields
    specialization: str | None = None
    department: str | None = None
    bio: str | None = None
    consultation_fee: Decimal | None = None
    languages_spoken: list[str] = []
    slot_duration_minutes: int = 30
    buffer_minutes: int = 0


class DoctorUpdate(BaseModel):
    specialization: str | None = None
    department: str | None = None
    bio: str | None = None
    consultation_fee: Decimal | None = None
    languages_spoken: list[str] | None = None
    slot_duration_minutes: int | None = None
    buffer_minutes: int | None = None


class DoctorResponse(BaseModel):
    id: UUID
    specialization: str | None
    department: str | None
    bio: str | None
    consultation_fee: Decimal | None
    languages_spoken: list[str]
    slot_duration_minutes: int
    buffer_minutes: int
    is_active: bool
    user: UserResponse

    model_config = {"from_attributes": True}


class DoctorPublicProfile(BaseModel):
    """Returned to patients browsing doctors — no internal fields."""
    id: UUID
    specialization: str | None
    department: str | None
    bio: str | None
    consultation_fee: Decimal | None
    languages_spoken: list[str]
    slot_duration_minutes: int
    name: str       # flattened from user
    avatar_url: str | None

    model_config = {"from_attributes": True}
